src/ast.rs

```diff
@@ -0,0 +1,234 @@
+                        "{: >1$}",
+                        "",
+                        self.number_of_spaces * self.indents
+                    ))?;
+                }
+
+                self.writer.write_str(line)?;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+proxy_display!(ReturnStatement);
+
+#[derive(Debug)]
+pub(crate) struct ReturnStatement(pub Expression, pub ASTTarget);
+
+impl ASTFormat for ReturnStatement {
+    fn fmt_ast(&self, output: &mut ASTStream) -> fmt::Result {
+        match output.target {
+            ASTTarget::Lua => {
+                write!(output, "return ")
+            }
+            ASTTarget::Typescript { output_dir } => {
+                write!(output, "declare const {output_dir}: ")
+            }
+        }
+        .expect("Failed to write to output");
+        let result = self.0.fmt_ast(output);
+        if let ASTTarget::Typescript { output_dir } = output.target {
+            write!(output, "\nexport = {output_dir}")?
+        }
+        result
+    }
+}
+
+#[derive(Debug)]
+pub(crate) enum Expression {
+    String(String),
+    Table(Table),
+}
+
+impl Expression {
+    pub fn table(expressions: Vec<(Expression, Expression)>) -> Self {
+        Self::Table(Table { expressions })
+    }
+}
+
+impl ASTFormat for Expression {
+    fn fmt_ast(&self, output: &mut ASTStream) -> fmt::Result {
+        match self {
+            Self::Table(val) => val.fmt_ast(output),
+            Self::String(val) => val.fmt_ast(output),
+        }
+    }
+
+    fn fmt_key(&self, output: &mut ASTStream<'_, '_>) -> fmt::Result {
+        match self {
+            Self::Table(val) => val.fmt_key(output),
+            Self::String(val) => val.fmt_key(output),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct Table {
+    pub expressions: Vec<(Expression, Expression)>,
+}
+
+impl ASTFormat for Table {
+    fn fmt_ast(&self, output: &mut ASTStream<'_, '_>) -> fmt::Result {
+        let assignment = match output.target {
+            ASTTarget::Lua => " = ",
+            ASTTarget::Typescript { .. } => ": ",
+        };
+
+        writeln!(output, "{{")?;
+        output.indent();
+
+        for (key, value) in &self.expressions {
+            key.fmt_key(output)?;
+            write!(output, "{assignment}")?;
+            value.fmt_ast(output)?;
+            writeln!(output, ",")?;
+        }
+
+        output.unindent();
+        write!(output, "}}")
+    }
+}
+
+impl ASTFormat for String {
+    fn fmt_ast(&self, output: &mut ASTStream) -> fmt::Result {
+        write!(output, "\"{}\"", self)
+    }
+
+    fn fmt_key(&self, output: &mut ASTStream<'_, '_>) -> fmt::Result {
+        if is_valid_identifier(self) {
+            write!(output, "{}", self)
+        } else {
+            match output.target {
+                ASTTarget::Lua => write!(output, "[\"{}\"]", self),
+                ASTTarget::Typescript { .. } => write!(output, "\"{}\"", self),
+            }
+        }
+    }
+}
```

Comment:
`AST` should be `Ast`

Reply:
Should be resolved in: https://github.com/jackTabsCode/asphalt/pull/33/commits/65076f6a46b684dfd3b5b1a965e005c626c4e199